logicrunner/logicrunner.go

```diff
@@ -352,12 +352,8 @@ func (lr *LogicRunner) Execute(ctx context.Context, parcel core.Parcel) (core.Re
 		return nil, errors.Wrap(err, "can't play role")
 	}
 
-	if es.Current != nil {
-		// TODO: check no wait call
-		if inslogger.TraceID(es.Current.Context) == inslogger.TraceID(ctx) {
-			es.Unlock()
-			return nil, os.WrapError(nil, "loop detected")
-		}
+	if loop := lr.CheckExecutionLoop(ctx, es, parcel); loop {
```

Comment:
Why do you assign result of `CheckExecutionLoop ` to `loop` and never use it?

Reply:
shit :) I use it right after ";" on the same line ^))))))))))

Reply:
"fixed"